addons/sale/models/sale_order.py

```diff
@@ -1291,6 +1291,7 @@ def _default_order_line_values(self, child_field=False):
     def _get_action_add_from_catalog_extra_context(self):
         return {
             **super()._get_action_add_from_catalog_extra_context(),
+            'display_uom': self.env.user.has_group('uom.group_uom'),
```

Comment:
It would be simpler to only include `uom` if the group is enabled, that way no need to do anycheck, you show `uom` only if it's there.